mlx/directives/item_directive.py

```diff
@@ -51,7 +53,7 @@ def perform_replacement(self, app, collection):
                         report_warning(env,
                                        'Traceability: relation {rel} cannot be translated to string'
                                        .format(rel=rel),
-                                       self.docname, self.line)
+                                       env.docname, self.line)
```

Comment:
are we sure about this? Sounds like now warning might be on some other docname, or we simply do not init docname in class anymore?

Reply:
I am sure. I tried to init docname in the class as part of the refactoring, but it turned out to be of no use, so I reverted all but this instance. I should've reverted this on the other branch already.